### ISSUE TITLE
Resetting forms

### ISSUE THREAD

**Issue by matthew-white** [getodk/central-frontend#193]
Right now we tend not to reset a form when the user leaves the form. A few examples:

1. If the user navigates to the project app users page, clicks "Create App User", enters a nickname, then clicks Cancel, the form does not reset: if the user clicks "Create App User" to reopen the modal, the nickname that was entered is still there.
2. If the user navigates to the project overview page, clicks "App Users", clicks "Create App User", enters a nickname, then presses the Back button to return to the project overview, the form does not reset: if the user clicks "App Users", then clicks "Create App User" to reopen the modal, the nickname that was entered is still there.
3. If the user navigates to the form attachments page, drops a file onto the page, then clicks "Settings" to navigate to the form settings page, the form attachments page does not reset: if the user clicks "Media Files" to return to the form attachments page, the popup is still there.

My sense is that it's unexpected in these cases that the form or page does not reset, especially in cases (2) and (3). However, my guess is that implementing this sort of reset would require additional logic, and I'm not sure that the current behavior is unexpected enough that we'd want to add that.

@clint-tseng, do you have preferences about this sort of behavior? Do you think it'd be worth implementing this sort of reset?

**Comment by clint-tseng** [comment]
yeah, reset would probably be preferable. but i don't think it's the most critical thing at the moment either.

**Comment by matthew-white** [comment]
We've decided to reset modals after they are hidden, which should resolve examples (1) and (2). We have made progress on resetting modals with #203 (af8e7fc6e79076c7f02f915c95105d2dcb8d8a9f).

**Comment by matthew-white** [comment]
I took a quick look, and I think we now reset forms everywhere:

- We no longer use `<keep-alive>`.
- As of #382, we do not reuse a component after a route update.
- All modals seem to be reset when they are hidden.
- I also looked at `<form>`, `<input>`, and `<select>` elements, as well as other elements with an attached submit event listener, and things seem to be reset.

If I've missed a form control somewhere, I can fix that as a one-off, but I'm pretty sure I've caught everything, so I'll go ahead and close this issue.
